src/Components/EntityState.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Components
+{
+	public interface EntityState
+	{
+		bool TryAdd<Component>(Component component);
+		Component Get<Component>();
```

Comment:
That's fair, and makes sense, I did think that was more towards what you was aiming for but wanted to throw this out there anyway